src/sidebar/services/persisted-defaults.js

```diff
@@ -45,10 +43,9 @@ export default function persistedDefaults(localStorage, store) {
         const defaultValue = localStorage.getItem(DEFAULT_KEYS[defaultKey]);
         store.setDefault(defaultKey, defaultValue);
       });
-      lastDefaults = store.getDefaults();
 
       // Listen for changes to those defaults from the store and persist them
-      store.subscribe(persistChangedDefaults);
+      watch(store.subscribe, () => store.getDefaults(), persistChangedDefaults);
```

Comment:
Yes, it currently will. It creates a refactoring hazard though because we're passing unused arguments to `getDefaults`. If `getDefaults` was changed to have optional arguments in future, the values passed by `watch` could have unexpected behavior. This is especially true since the arguments are not type-checked. This is why in various places we have closures like `() => someCallback()` to discard unused arguments.